diamond.py

```diff
@@ -1951,11 +1951,20 @@ def generalised_kummer(n):
         sage: all(generalised_kummer(n).betti()[2] == 7 for n in range(3, 10))
         True
     """
-    def product(n):
-        x = HodgeDiamond.x
-        y = HodgeDiamond.y
+    x = HodgeDiamond.x
+    y = HodgeDiamond.y
 
-        return HodgeDiamond.R(sum([gcd(list(a.keys()))**4 * (x*y)**(n - sum(a.values())) * prod([sum([prod([1 / HodgeDiamond.R((j**b[j]) * factorial(b[j])) * ((1-x**j) * (1-y**j))**(2*b[j]) for j in b.keys()]) for b in [b.to_exp_dict() for b in Partitions(a[i])]]) for i in a.keys()]) for a in [a.to_exp_dict() for a in Partitions(n)]])(-x, -y))
+    def my_Partitions(k):
+        for pi in Partitions(k):
+            yield pi.to_exp_dict()
+
+    def product(n):
+        hd = sum(gcd(a)**4 * (x*y)**(n - sum(a.values())) *
+                 prod([sum([prod([~((j**bj) * factorial(bj)) *
+                                  ((1 - x**j) * (1 - y**j))**(2*bj) for j, bj in b.items()])
```

Comment:
Why not write `b.to_exp_dict().items()` on this line, and avoid `my_Partitions(k)` altogether?

Reply:
Fair enough, on the next line it would be `a.to_exp_dict().values()`.

Reply:
the use of `my_Permutations` is a simple way to run through partitions without to ever create a list of partitions

Reply:
But replacing it with a call to Partitions and performing `to_exp_dict()` inline would also not create the full list of partitions, or am I mistaken?

Reply:
one could do what you suggest for b, but not for a, because a is called three times